src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1552,4 +1545,127 @@ public int cullInvalidConsumerContentAccess(Owner owner, String... existingModes
+        if (certIds == null || certIds.isEmpty()) {
+            return 0;
+        }
+
+        String query = "" +
+            "UPDATE cp_consumer" +
+            " SET cont_acc_cert_id = NULL, updated = :date" +
+            " FROM cp_consumer a, cp_cont_access_cert c" +
+            " WHERE a.cont_acc_cert_id = c.id" +
+            " AND c.id IN (:cert_ids)";
+
+        return this.currentSession().createSQLQuery(query)
+            .setParameter("date", new Date())
+            .setParameter("cert_ids", certIds)
+            .executeUpdate();
+    }
+
+    @Transactional
+    private int mariaDbUnlinkCaCertificates(Collection<String> certIds) {
+        if (certIds == null || certIds.isEmpty()) {
+            return 0;
+        }
+
+        String query = "" +
+            "UPDATE cp_consumer a" +
+            " JOIN cp_cont_access_cert c ON a.cont_acc_cert_id = c.id " +
+            " SET a.cont_acc_cert_id = NULL, a.updated = :date" +
+            " WHERE c.id IN (:cert_ids)";
+
+        return this.currentSession().createSQLQuery(query)
+            .setParameter("date", new Date())
+            .setParameter("cert_ids", certIds)
+            .executeUpdate();
+    }
+
+    /**
+     * Takes a list of identity certificate id and unlinks them from consumers.
+     *
+     * @param certIds certificate ids to be unlinked
+     * @return a number of unlinked consumers
+     * @throws PersistenceException when unsupported db dialect is detected
+     */
+    @Transactional
+    public int unlinkIdCertificates(Collection<String> certIds) throws PersistenceException {
+        String db = getDbDialect();
+        if (db.contains("mysql") || db.contains("maria")) {
+            return this.mariaDbUnlinkIdCertificates(certIds);
+        }
+        else if (db.contains("postgresql")) {
+            return this.postgreUnlinkIdCertificates(certIds);
+        }
+        else {
+            throw new PersistenceException(
+                "Cannot execute query: The database dialect is not recognized.");
+        }
+    }
+
+    /**
+     * Takes a list of content access certificate id and unlinks them from consumers.
+     *
+     * @param certIds certificate ids to be unlinked
+     * @return a number of unlinked consumers
+     * @throws PersistenceException when unsupported db dialect is detected
+     */
+    @Transactional
+    public int unlinkCaCertificates(Collection<String> certIds) throws PersistenceException {
+        String db = getDbDialect();
+        if (db.contains("mysql") || db.contains("maria")) {
+            return this.mariaDbUnlinkCaCertificates(certIds);
+        }
+        else if (db.contains("postgresql")) {
+            return this.postgreUnlinkCaCertificates(certIds);
+        }
+        else {
+            throw new PersistenceException(
+                "Cannot execute query: The database dialect is not recognized.");
+        }
+    }
+
+    private String getDbDialect() {
```

Comment:
If we're going to abstract this out, move it up a tier to the AbstractHibernateCurator.

Reply:
Done.